core/src/frame_lifecycle.rs

```diff
@@ -0,0 +1,169 @@
+//! Frame events management
+//!
+//! This module aids in keeping track of which frame execution phase we are in.
+//!
+//! For AVM2 code, display objects execute a series of discrete phases, and
+//! each object is notified about the current frame phase in rendering order.
+//! When objects are created, they are 'caught up' to the current frame phase
+//! to ensure correct order of operations.
+//!
+//! AVM1 code (presumably, either on an AVM1 stage or within an `AVM1Movie`)
+//! runs in one phase, with timeline operations executing with all phases
+//! inline in the order that clips were originally created.
+
+use crate::context::UpdateContext;
+use crate::display_object::{DisplayObject, TDisplayObject};
+use crate::vminterface::AvmType;
+
+/// Which phase of the frame we're currently in.
+///
+/// Each part of the frame phase is the phase we're going to execute *next*;
+/// e.g. we don't go from `Enter` to `Construct`
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum FramePhase {
+    /// We're about to enter the next frame.
+    ///
+    /// When we enter a new frame, movie clips increment their current frame
+    /// number. Once this phase ends, we fire `enterFrame` on the broadcast
+    /// list.
+    ///
+    /// When the player is not executing a frame advancing update, it should be
+    /// in this phase.
```

Comment:
Is the comment correct here? Shouldn't it be in `Idle` by default? (and the code feels like it too)

Reply:
You're correct. I added the `Idle` phase after writing the doccomments.